File: cozy_android/src/main/java/ua/android/cozy/cozyandroid/mvp/ui/NavigateActivity.java
package ua.android.cozy.cozyandroid.mvp.ui;

import android.support.v4.app.FragmentManager;
import android.view.View;
import android.widget.FrameLayout;

import ua.android.cozy.cozyandroid.navigator.Navigator;

/**
 * Created by Palamarenko Andrey on
 * 31.03.2018 11:46
 */

public class NavigateActivity extends BaseActivity {

    private Navigator navigator;


    public void simpleInit(){
        FrameLayout frameLayout = new FrameLayout(this);
        frameLayout.setId(View.generateViewId());
        setContentView(frameLayout);
        navigator = new Navigator(frameLayout.getId(),getSupportFragmentManager());
    }


    public Navigator getNavigator() {
        return navigator;
    }


    @Override
    public void onBackPressed() {
        FragmentManager manager = getNavigator().getFragmentManager();
        int count = manager.getBackStackEntryCount();
        if (count == 1) {
            finish();
        } else {
            super.onBackPressed();
        }
    }
}

File: cozy_android/src/main/java/ua/android/cozy/cozyandroid/viewlayer/NavigateActivity.kt
package ua.android.cozy.cozyandroid.viewlayer

import android.os.Bundle
import android.support.v4.app.FragmentManager
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.FrameLayout

import ua.android.cozy.cozyandroid.navigator.Navigator

open class NavigateActivity : AppCompatActivity() {

    lateinit var navigator: Navigator

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        simpleInit()
    }

    fun simpleInit() {
        val frameLayout = FrameLayout(this)
        frameLayout.id = View.generateViewId()
        setContentView(frameLayout)
        navigator = Navigator(frameLayout.id, supportFragmentManager)
    }


    override fun onBackPressed() {
        val manager = navigator.fragmentManager
        val count = manager.backStackEntryCount
        if (count == 1) {
            finish()
        } else {
            super.onBackPressed()
        }
    }
}
